src/Assert.php

```diff
@@ -904,6 +1015,11 @@ public static function classExists($value, $message = '')
         }
     }
 
+    /**
+     * @psalm-template ExpectedType of object
+     * @psalm-param class-string<ExpectedType> $class
+     * @psalm-assert ExpectedType|class-string<ExpectedType> $value
```

Comment:
this isn't supported yet - there'll need to be an annotation like `object-or-class-string<ExpectedType>`